lib/qmi/codec/wireless_data.ex

```diff
@@ -68,4 +103,128 @@ defmodule QMI.Codec.WirelessData do
+  """
+  @spec parse_indication(binary()) ::
+          {:ok, packet_status_indication()} | {:error, :invalid_indication}
+  def parse_indication(
+        <<@packet_service_status_ind::16-little, size::16-little, tlvs::binary-size(size)>>
+      ) do
+    result = parse_packet_status_indication(packet_status_indication_init(), tlvs)
+
+    {:ok, result}
+  end
+
+  def parse_indication(_other) do
+    {:error, :invalid_indication}
+  end
+
+  defp packet_status_indication_init() do
+    %{
+      status: nil,
+      reconfiguration_required: nil,
+      call_end_reason: nil,
+      call_end_reason_type: nil,
+      ip_family: nil,
+      tech_name: nil,
+      bearer_id: nil,
+      xlat_capable: nil
+    }
+  end
+
+  defp parse_packet_status_indication(indication, <<>>) do
+    indication
+  end
+
+  defp parse_packet_status_indication(
+         indication,
+         <<0x01, 0x02, 0x00, connection_status, reconfig_required, rest::binary>>
+       ) do
+    indication
+    |> Map.put(:status, parse_packet_status(connection_status))
+    |> Map.put(:reconfiguration_required, reconfig_required == 1)
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(
+         indication,
+         <<0x10, 0x02, 0x00, call_end_reason::16-little, rest::binary>>
+       ) do
+    indication
+    |> Map.put(:call_end_reason, call_end_reason)
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(
+         indication,
+         <<0x11, 0x04, 0x00, call_end_reason_type::16-little, call_end_reason::16-little,
+           rest::binary>>
+       ) do
+    indication
+    |> Map.put(:call_end_reason_type, parse_call_end_reason_type(call_end_reason_type))
+    |> Map.put(:call_end_reason, call_end_reason)
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(indication, <<0x12, 0x01, 0x00, ip_family, rest::binary>>) do
+    indication
+    |> Map.put(:ip_family, ip_family)
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(
+         indication,
+         <<0x13, 0x02, 0x00, tech_name::16-little, rest::binary>>
+       ) do
+    indication
+    |> Map.put(:tech_name, parse_tech_name(tech_name))
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(indication, <<0x14, 0x01, 0x00, bearer_id, rest::binary>>) do
+    indication
+    |> Map.put(:bearer_id, bearer_id)
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(
+         indication,
+         <<0x15, 0x01, 0x00, xlat_capable, rest::binary>>
+       ) do
+    indication
+    |> Map.put(:xlat_capable, xlat_capable == 1)
+    |> parse_packet_status_indication(rest)
+  end
+
+  defp parse_packet_status_indication(
+         indication,
+         <<_type, length::16-little, _values::binary-size(length), rest::binary>>
+       ) do
+    parse_packet_status_indication(indication, rest)
```

Comment:
Maybe we should be noisy here and blast that we don't know what these bytes of the indication is? Or maybe this intentional since it's all the documentation we can find and won't really know what else can be here..

Reply:
Yeah, this is mostly in place to avoid surprise TLVs that we don't support. Maybe adding a log would be helpful, but even so, if we cannot docs on those values we won't really know how to provide meaning to them. However, with what we do have I supported all fields regardless if they seemed necessary for the problem we are trying to solve right now, so we could delete this catch-all.